scripts/build/bootstrap/incremental_build_util.py

```diff
@@ -230,24 +268,25 @@ def create_volume(ec2_client, availability_zone, repository_name, project, pipel
     time.sleep(1)
 
     response = ec2_client.describe_volumes(VolumeIds=[volume_id, ])
-    while (response['Volumes'][0]['State'] != 'available'):
-        time.sleep(1)
-        response = ec2_client.describe_volumes(VolumeIds=[volume_id, ])
+    with timeout(DEFAULT_TIMEOUT, 'ERROR: Timeout reached trying to create EBS.'):
```

Comment:
Is a timeout necessary when this is executed from Jenkins and that has already a timeout?
In case this part fails, is there something that can save the script from failing?

The change here just enables to kill the script while this is iterating, right?

Reply:
This timeout threshold is less than Jenkins timeout, so we can fail fast.
The timeout is useful especially when we do unmounting, we unmount EBS from device with force=False, it it timed out to unmount, then we will force unmount the EBS.

If attaching or mounting an EBS timed out, I think we don't need to keep the script running because we won't have a EBS volume to run the build, but yes, we can save the build from failing if detaching or unmounting failed.